python/whylogs/core/constraints/factories/cardinality_metrics.py

```diff
@@ -0,0 +1,30 @@
+from whylogs.core.metrics.metrics import Metric
+
+from ..metric_constraints import MetricConstraint, MetricsSelector
+
+
+def distinct_number_in_range(column_name: str, lower: int, upper: int) -> MetricConstraint:
+    """Number of distinct categories must be between lower and upper values (inclusive).
+
+    Parameters
+    ----------
+    column_name : str
+        Column the constraint is applied to
+    lower : int
+        Lower bound of defined range
+    upper : int
+        Upper bound of the value range
+    """
+
+    def distinct_in_range(metric: Metric):
+        distinct_estimate = metric.hll.value.get_estimate()
```

Comment:
if the hll metric is empty does this throw?

Reply:
@jamie256 in the tests i ran, all column types will lead to existing hll metrics. talking to felipe we were not sure whether this was correct or not

Reply:
![image](https://user-images.githubusercontent.com/48736656/181637806-32e8138a-8d83-4cfc-ba32-7bd981dd412f.png)
this was from a 10 elements np.float type pandas DataFrame.